x-pack/plugins/licensing/public/license.test.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ILicense } from '../common/types';
+import { LICENSE_STATUS } from '../common/constants';
+import { LicenseFeature } from '../common/license_feature';
+import { setup } from './__fixtures__/setup';
+import { Plugin } from './plugin';
+
+describe('license', () => {
+  let plugin: Plugin;
+  let license: ILicense;
+
+  afterEach(async () => {
+    await plugin.stop();
+    sessionStorage.clear();
+  });
+
+  test('uid returns a UID field', async () => {
+    ({ plugin, license } = await setup());
+
+    expect(license.uid).toBe('00000000-0000-0000-0000-000000000000');
+  });
+
+  test('isActive returns true if status is active', async () => {
+    ({ plugin, license } = await setup());
+
+    expect(license.isActive).toBe(true);
+  });
+
+  test('isActive returns false if status is not active', async () => {
+    ({ plugin, license } = await setup({
+      license: {
+        status: 'aCtIvE', // needs to match exactly
+      },
+    }));
+
+    expect(license.isActive).toBe(false);
+  });
+
+  test('expiryDateInMillis returns expiry_date_in_millis', async () => {
+    const expiry = Date.now();
+
+    ({ plugin, license } = await setup({
+      license: {
+        expiry_date_in_millis: expiry,
+      },
+    }));
+
+    expect(license.expiryDateInMillis).toBe(expiry);
+  });
+
+  test('isOneOf returns true if the type includes one of the license types', async () => {
+    ({ plugin, license } = await setup({
+      license: {
+        type: 'platinum',
+      },
+    }));
+
+    expect(license.isOneOf('platinum')).toBe(true);
+    expect(license.isOneOf(['platinum'])).toBe(true);
+    expect(license.isOneOf(['gold', 'platinum'])).toBe(true);
+    expect(license.isOneOf(['platinum', 'gold'])).toBe(true);
+    expect(license.isOneOf(['basic', 'gold'])).toBe(false);
+    expect(license.isOneOf(['basic'])).toBe(false);
+  });
+
+  test('type returns the license type', async () => {
+    ({ plugin, license } = await setup());
+
+    expect(license.type).toBe('basic');
+  });
+
+  test('returns feature API with getFeature', async () => {
+    ({ plugin, license } = await setup());
+
+    const security = license.getFeature('security');
+    const fake = license.getFeature('fake');
+
+    expect(security).toBeInstanceOf(LicenseFeature);
+    expect(fake).toBeInstanceOf(LicenseFeature);
+  });
+
+  describe('isActive', () => {
+    test('should return Valid if active and check matches', async () => {
+      ({ plugin, license } = await setup({
+        license: {
+          type: 'gold',
+        },
+      }));
+
+      expect(license.check('test', 'basic').check).toBe(LICENSE_STATUS.Valid);
```

Comment:
- a case with number is not tested. 
- is it possible to narrow down interface to accept either strings only?

Reply:
It is, but I'd like to address this in a follow up since it would affect all of licensing and not just the public bits.